proxy/src/main/java/com/wavefront/agent/handlers/AbstractSenderTask.java

```diff
@@ -185,8 +187,18 @@ protected List<T> createBatch() {
     List<T> current;
     int blockSize;
     synchronized (mutex) {
+      int rateLimit = (int)rateLimiter.getRate();
```

Comment:
Add comment that you considered storing the bytes and opted for the method of least code change to start and should we hit cpu issues, this is a potential area of re-visit